### PR TITLE
Pin third-party actions to hashes

### PR DESCRIPTION
**Pull request by StevenMaude**
Pin all the non-GitHub supplied actions.

### CODE REVIEW

**Review by ghickman — COMMENTED** [review]
Can you pop the version on the end as a comment too?  We've seen Dependabot will update those too!

**Review by ghickman — APPROVED** [review]

**Comment by StevenMaude** [comment]
@ghickman: didn't know that, can you link an example, please? :grin:

File: .github/workflows/check_links.yml [hunk 1]
```diff
@@ -18,7 +18,7 @@ jobs:
         uses: actions/checkout@v3
 
       - name: Check links
-        uses: lycheeverse/lychee-action@v1.5.4
+        uses: lycheeverse/lychee-action@4dcb8bee2a0a4531cba1a1f392c54e8375d6dd81
```
Comment (bloodearnest) [inline .github/workflows/check_links.yml:21]:
```suggestion
        uses: lycheeverse/lychee-action@4dcb8bee2a0a4531cba1a1f392c54e8375d6dd81  # v1.5.4
```

File: .github/workflows/check_snippets.yml [hunk 2]
```diff
@@ -19,7 +19,7 @@ jobs:
           cache: "pip"
           cache-dependency-path: databuilder/requirements.*.txt
 
-      - uses: extractions/setup-just@v1
+      - uses: extractions/setup-just@95b912dc5d3ed106a72907f2f9b91e76d60bdb76
```
Comment (bloodearnest) [inline .github/workflows/check_snippets.yml:22]:
```suggestion
      - uses: extractions/setup-just@95b912dc5d3ed106a72907f2f9b91e76d60bdb76  # v1.5
```

File: .github/workflows/pages-deployment.yml [hunk 3]
```diff
@@ -32,7 +32,7 @@ jobs:
 
       - name: Publish
         if: ${{ github.actor != 'dependabot[bot]' }}
-        uses: cloudflare/pages-action@1
+        uses: cloudflare/pages-action@c4f0d2e08d902c9cdfe3cdf854896c6f363ae5be
```
Comment (bloodearnest) [inline .github/workflows/pages-deployment.yml:35]:
```suggestion
        uses: cloudflare/pages-action@c4f0d2e08d902c9cdfe3cdf854896c6f363ae5be  # v1.1.0
```

**Review by bloodearnest — APPROVED** [review]
Have added suggestions to record the semantic version in a comment, for future travellers

**Comment by StevenMaude** [comment]
:magic_wand: thank you both :+1:

**Comment by ghickman** [comment]
@StevenMaude – Simon has covered the format, so just to add to this: https://github.com/opensafely-core/job-server/pull/2405/files Look! Dependabot! updating the comment!

**Pull request merged**